### ISSUE TITLE
Implementing defer 

### ISSUE THREAD

**Issue by dgurns** [jacob-ebey/remix-d1-northwind#1]
hey Jacob, been following and enjoying your amazing work on `defer`. Thank you. 

I was upgrading a Remix app to `1.11.1` today which also uses Cloudflare and D1. https://github.com/dgurns/airish-tunes

From the Remix docs, it wasn't clear how to update my `entry.client.tsx` and `entry.server.tsx` files after updating to `remix@1.11.1` and `react/react-dom@18.2.0` in order to support streaming and hydration.

Luckily I found the answer here in your repo - https://github.com/jacob-ebey/remix-d1-northwind/blob/main/app/entry.server.tsx

I'm happy to do a PR to add this to the Remix docs, but where do you think is the best place to add that? 

Basically it would have been helpful to have a guide for "How to upgrade your app to support defer"
- Update remix, react, react-dom
- Update the entry files

Looks like the Remix Cloudflare Pages app template also needs to be updated to have the new entry file contents. 

Thanks again for all your work on this, it is super cool.

**Comment by dgurns** [comment]
Opened a PR to update the templates: https://github.com/remix-run/remix/pull/5213
